spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java

```diff
@@ -23,6 +23,9 @@
             "java.net.URI", "java.net.Inet4Address", "java.net.Inet6Address", "java.net.InetSocketAddress",
             "java.security.Permission"));
 
+    private static final Set<String> KNOWN_IMMUTABLE_PACKAGES = new HashSet<>(Arrays.asList(
```

Comment:
Could some variant on this be used for the Guava immutable classes? Perhaps list prefixes (eg `com.google.common.collect.Immutable`) instead of package names?

Reply:
It might be more efficient, is all.